depthai_examples/ros1_src/stereo_inertial_publisher.cpp

```diff
@@ -374,27 +376,47 @@ int main(int argc, char** argv) {
                                                         nnPath);
 
     std::shared_ptr<dai::Device> device;
-    std::vector<dai::DeviceInfo> availableDevices = dai::Device::getAllAvailableDevices();
-
-    std::cout << "Listing available devices..." << std::endl;
-    for(auto deviceInfo : availableDevices) {
-        std::cout << "Device Mx ID: " << deviceInfo.getMxId() << std::endl;
-        if(deviceInfo.getMxId() == mxId) {
-            if(deviceInfo.state == X_LINK_UNBOOTED || deviceInfo.state == X_LINK_BOOTLOADER) {
-                isDeviceFound = true;
-                if(poeMode) {
-                    device = std::make_shared<dai::Device>(pipeline, deviceInfo);
-                } else {
-                    device = std::make_shared<dai::Device>(pipeline, deviceInfo, usb2Mode);
+    if (useWithIP) // Connecting to a camera with specific IP
+    {
+        auto deviceInfo = dai::DeviceInfo(ipAddress);
+        if(deviceInfo.state == X_LINK_UNBOOTED || deviceInfo.state == X_LINK_BOOTLOADER || deviceInfo.state == X_LINK_FLASH_BOOTED) {
+            isDeviceFound = true;
+            std::cout << "Device found with IP Address: " << ipAddress <<  std::endl;
+            if(poeMode) {
+                device = std::make_shared<dai::Device>(pipeline, deviceInfo);
+            } else {
+                device = std::make_shared<dai::Device>(pipeline, deviceInfo, usb2Mode);
+            }
+        } else if(deviceInfo.state == X_LINK_BOOTED) {
+            throw std::runtime_error("ros::NodeHandle() from Node \"" + pnh.getNamespace() + "\" DepthAI Device with ipAddress  \"" + ipAddress
+                                     + "\" is already booted on different process.  \"");
+        } else {
```

Comment:
New update fixes the setting the MxID in depthai-core. So why not use that here ?
IP address can change based on DHPC in the setup right ?

Reply:
Oh really? That is great news. At the time I was testing MxID it was not working, so as a workaround I assigned static IP addresses to each camera and connect in this way. I feel like having an option to connect to specific IP address might be useful for some users, so what I will be do is add another parameter like useMxID, users can specify whether they wanna use IP, mxID or default (automatically connecting). What do you think about this plan?

Reply:
Yup. Sounds good. Thanks.